python-checks/src/test/resources/pandasModifyInPlace.py

```diff
@@ -0,0 +1,95 @@
+def non_compliant_1():
+    import pandas as pd
+
+    df = pd.DataFrame({'A': [3, 2, 1], 'B': ['x', 'y', 'z']})
+    df2 = pd.DataFrame({'A': [6, 5, 4], 'C': ['a', 'b', 'c']})
+
+    new_index = ['Firefox', 'Chrome', 'Safari']
+
+    _ = df.drop(columns='A', inplace=True)  # Noncompliant
+    #                        ^^^^^^^^^^^^
+
+    _ = df.dropna(inplace=True)  # Noncompliant
+    #             ^^^^^^^^^^^^
+
+    _ = df.drop_duplicates(inplace=True)  # Noncompliant
+    #                      ^^^^^^^^^^^^
+
+    _ = df.sort_values(by=['A'], inplace=True)  # Noncompliant
+    #                            ^^^^^^^^^^^^
+
+    _ = df.sort_index(inplace=True)  # Noncompliant
+    #                 ^^^^^^^^^^^^
+
+    _ = df.eval("C = str(A) + B", inplace=True)  # Noncompliant
+    #                             ^^^^^^^^^^^^
+
+    _ = df.query('A > B', inplace=True)  # Noncompliant
```

Comment:
I think it would be nice to also have a test that shows this work when the data frame is created through something else than the constructor, such as:
```
my_df = pd.read_csv("my_csv.csv")
    _ = my_df.query('A > B', inplace=True)  # Noncompliant
```